kratos/includes/checks.h

```diff
@@ -27,6 +27,9 @@
 #define KRATOS_CHECK_EQUAL(a,b) if(!(a == b)) KRATOS_ERROR << "Check failed because " << #a << " is not equal to " << #b
 #define KRATOS_CHECK_NOT_EQUAL(a,b) if(a == b) KRATOS_ERROR << "Check failed because " << #a << " is equal to " << #b
 
+#define KRATOS_CHECK_STRING_EQUAL(a,b) if(a.compare(b) != 0) KRATOS_ERROR << "Check failed because \"" << a << "\" is not equal to \"" << b << "\""
```

Comment:
just FYI, compare is the same as == for std::strings

Reply:
The problem was the output of KRATOS_ERROR was "#a" instead of "a".